drizzlepac/haputils/catalog_utils.py

```diff
@@ -1194,17 +1194,24 @@ def measure_sources(self, filter_name):
 
         try:
             # Calculate and add concentration index (CI) column to table
-            ci_data = photometry_tbl["MagAp1"].data - photometry_tbl["MagAp2"].data
-        except Exception:
-            log.info("Wrote catalog info to file 'catalog.pickle'.")
-            pickle_out = open("catalog.pickle", "wb")
-            pickle.dump(photometry_tbl, pickle_out)
-            pickle_out.close()
-
-        ci_mask = np.logical_and(np.abs(ci_data) > 0.0, np.abs(ci_data) < 1.0e-30)
-        big_bad_index = np.where(abs(ci_data) > 1.0e20)
-        ci_mask[big_bad_index] = True
-        ci_col = MaskedColumn(name="CI", data=ci_data, dtype=np.float64, mask=ci_mask)
+            if math.isclose(self.image.keyword_dict['photflam'], 0.0, abs_tol=1e-9):
+                # Fill the ci_data column with the "bad data indicator" of -9999.0
+                # where photometry_tbl["MagAp1"] was populated with -9999.0 in iraf_style_photometry
+                ci_data = photometry_tbl["MagAp1"].data
+                ci_mask = np.logical_not(ci_data > -9999.0)
+            else:
+                ci_data = photometry_tbl["MagAp1"].data - photometry_tbl["MagAp2"].data
+                ci_mask = np.logical_and(np.abs(ci_data) > 0.0, np.abs(ci_data) < 1.0e-30)
+                big_bad_index = np.where(abs(ci_data) > 1.0e20)
+                ci_mask[big_bad_index] = True
+
+        except Exception as x_cept:
+            log.warning("Computation of concentration index (CI) was not successful: {} - {}.".format(self.imgname, x_cept))
+            log.warning("CI measurements may be missing from the output filter catalog.\n")
+            ci_data = -9999.0 + photometry_tbl["MagAp1"] * 0.0
```

Comment:
Why not simply 'ci_data = -9999.0', since anything times zero is zero?

Reply:
I need the result to be a column and not a scalar.   I thought this would ensure I got a column as this is what the remainder of the code is expecting.

Reply:
That makes sense.  A comment to the effect would prevent confusion in the future.

Reply:
Addressed in 3a0707e.